src/workerd/util/sqlite.h

```diff
@@ -522,11 +544,64 @@ class SqliteDatabase::Query final: private ResetListener {
     }
   }
 
+  struct QueryEvent {
```

Comment:
This should be a class. Keep in mind everything is public in structs by default, so the setter methods aren't really enforceable unless your properties are designated `private`.